ivy/functional/backends/paddle/elementwise.py

```diff
@@ -719,9 +719,12 @@ def tanh(
     ]:
         return paddle.tanh(x.astype("float32")).astype(x.dtype)
     if paddle.is_complex(x):
-        tanh_a = paddle.tanh(paddle.real(x))
-        tan_b = paddle.tan(paddle.imag(x))
-        return (tanh_a + 1j * tan_b) / (1 + 1j * (tanh_a * tan_b))
+        const = paddle.to_tensor(1j, dtype=x.dtype)
+        tanh_a = paddle.cast(paddle.tanh(x.real()), x.dtype)
+        tan_b = paddle.cast(paddle.tan(x.imag()), x.dtype)
+        term1 = tanh_a + paddle.multiply(tan_b, const)
+        term2 = 1 + paddle.multiply(const, (tanh_a * tan_b))
+        return term1 / term2
```

Comment:
I'm not sure I see the reason for changing this.

Also, rather than casting a float to complex and then multiplying by 1j, it would probably be better to use the `paddle.complex(real, imag)` function as in:
```py
tanh_a = paddle.tanh(paddle.real(x))
tan_b = paddle.tan(paddle.imag(x))
term1 = paddle.complex(tanh_a, tan_b)
term2 = paddle.complex(1, tanh_a * tan_b)
return term1 / term2
```

Reply:
I have done that cuz paddle was giving me errors related to paddle functional API but I will modify the tanh implementation  as yo suggested.

Reply:
it seems there is a problem with paddle.complex not creating output if the real and imaginary part not of the same type
so I kept it the same before .
![image](https://github.com/unifyai/ivy/assets/97109954/a44c9945-ee14-437d-9221-1ccc0e722be2)

Reply:
It's likely the 1, you might need to do `paddle.ones_like(tanh_a)`